src/generators/typescript/TypeScriptGenerator.ts

```diff
@@ -39,15 +41,29 @@ export class TypeScriptGenerator extends AbstractGenerator<TypeScriptOptions,Typ
   ) {
     super('TypeScript', TypeScriptGenerator.defaultOptions, options);
   }
-  
+
   /**
    * Render a complete model result where the model code, library and model dependencies are all bundled appropriately.
    *
    * @param model
    * @param inputModel
    * @param options
    */
-  async renderCompleteModel(model: CommonModel, inputModel: CommonInputModel, options: TypeScriptRenderCompleteModelOptions): Promise<RenderOutput> {
+  async renderCompleteModel(model: CommonModel, inputModel: CommonInputModel, {moduleSystem = 'ESM', exportType = 'default'}: TypeScriptRenderCompleteModelOptions): Promise<RenderOutput> {
+    // Shallow copy presets so that we can restore it once we are done
+    const originalPresets = [...(this.options.presets ? this.options.presets : [])];
+
+    // Add preset that adds the `export` keyword if needed
+    if (
+      moduleSystem === 'ESM' &&
+      exportType === 'named' &&
+      (originalPresets[0] !== TS_EXPORT_KEYWORD_PRESET ||
+        (Object.prototype.hasOwnProperty.call(originalPresets[0], 'preset') &&
+          (originalPresets[0] as PresetWithOptions).preset !== TS_EXPORT_KEYWORD_PRESET))
+    ) {
+      this.options.presets = [TS_EXPORT_KEYWORD_PRESET, ...originalPresets];
+    }
```

Comment:
I see your problem yea 😅 I guess we are hitting the limits of what string presets can handle here.

Interesting workaround adding it as a preset and just ensuring it is the first one called 🤔 

I wonder if we cant make the check a bit easier to understand 🤔 
```
      (originalPresets[0] !== TS_EXPORT_KEYWORD_PRESET ||
        (Object.prototype.hasOwnProperty.call(originalPresets[0], 'preset') &&
          (originalPresets[0] as PresetWithOptions).preset !== TS_EXPORT_KEYWORD_PRESET))
```
Gonna think about that one for a bit.

cc @magicmatatjahu

Reply:
This bit detects the `presets: [TS_EXPORT_KEYWORD_PRESET]` syntax
 ```typescript
      (originalPresets[0] !== TS_EXPORT_KEYWORD_PRESET ||
```
This bit detects the `presets: [ { preset: TS_EXPORT_KEYWORD_PRESET, options?: { ... } } ]` syntax, that we don't add but the user might. I thought for completeness' sake it's better to include this and make it look a bit more complicated, but I'd be interested to hear what you think.
```typescript
        (Object.prototype.hasOwnProperty.call(originalPresets[0], 'preset') &&
          (originalPresets[0] as PresetWithOptions).preset !== TS_EXPORT_KEYWORD_PRESET))
 ```

Reply:
Might be just a matter of exporting this logic as a `hasPreset` or `isPreset` util function or something.

Reply:
I feel like this is just a bandage to the gun wound, cause eventually, we will end up in a situation where two presets cannot be used with each other because of how presets work 😅

That said, until that is fixed, I think this makes sense to introduce regardless 🤔

@magicmatatjahu @Samridhi-98 @gabormagyar what do you folks think?

Reply:
Yeah, I agree on this, it's just a temporary fix.

Reply:
So can we merge it?

Reply:
Left a comment 🙂